intake/catalog/base.py

```diff
@@ -276,6 +276,8 @@ def serialize(self):
             kw = entry._captured_init_kwargs.copy()
             kw.pop('catalog', None)
             kw['parameters'] = {k.name: k.__getstate__()['kwargs'] for k in kw.get('parameters', [])}
+            if not isinstance(kw['driver'], str):
+                kw['driver'] = str(kw['driver']).split("'")[1]
```

Comment:
Could be `".".join([driver.__module__, driver.__name__])` ? Actually, ":" should also work as the join character I think.

I think that, instead of doing this for all non-string things, I would change the check to
```
if issubclass(kw['driver'], intake.source.base.DataSourceBase)
```